### DISCUSSION TITLE
The example for ui.run_javascript() doesn't work correctly?

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#3472] [General]
I run the example code directly but got this error when click the button 'receive result'
```py
from nicegui import ui

def alert():
    ui.run_javascript('alert("Hello!")')

async def get_date():
    time = await ui.run_javascript('Date()')
    ui.notify(f'Browser time: {time}')

def access_elements():
    ui.run_javascript(f'getElement({label.id}).innerText += " Hello!"')

ui.button('fire and forget', on_click=alert)
ui.button('receive result', on_click=get_date)
ui.button('access elements', on_click=access_elements)
label = ui.label()

ui.run()
```
here is the error message
```
 File "D:\Program Files\python\Lib\site-packages\nicegui\client.py", line 222, in send_and_wait
    raise RuntimeError('Cannot await JavaScript responses on the auto-index page. '
RuntimeError: Cannot await JavaScript responses on the auto-index page. There could be multiple clients connected and it is not clear which one to wait for.
```

I found this 
`Raise exception when awaiting JavaScript on [auto-index page](https://nicegui.io/documentation/section_pages_routing) to avoid aftereffects with multiple connected clients (https://github.com/zauberzeug/nicegui/issues/2951, https://github.com/zauberzeug/nicegui/pull/2983 by @CrystalWindSnake, @frankvp11, @falkoschindler)`

but the example on the official site doesn't have such error message,should we change the example code to this?
```py
from nicegui import ui

def alert():
    ui.run_javascript('alert("Hello!")')

@ui.page('/time')
def time():
    async def get_date():
        time = await ui.run_javascript('Date()')
        ui.notify(f'Browser time: {time}')
    ui.button('receive result', on_click=get_date)

def access_elements():
    ui.run_javascript(f'getElement({label.id}).innerText += " Hello!"')

ui.button('fire and forget', on_click=alert)
ui.button('access elements', on_click=access_elements)
label = ui.label()

ui.run()
```

**Comment by python-and-novella** [comment]
you should only await code which belongs to `ui.page`.

**Reply by v479038280** [reply]
The official example code await outside of ui.page

**Comment by falkoschindler** [comment]
Yes, @v479038280, I noticed that too. The demos including `await ui.run_javascript(...)` only work because the whole demo is executed inside a `ui.page`, even though the displayed code doesn't show that. So the documentation is indeed misleading and should be updated.

**Reply by falkoschindler** [reply]
To be more precise: We should check every demo including `await [a-z]*\.run` (regex).

**Reply by falkoschindler** [reply]
See PR #3517.
